exasol/ds/sandbox/lib/dss_docker/push_image.py

```diff
@@ -0,0 +1,79 @@
+import docker
+import json
+import logging
+import requests
+
+from docker.client import DockerClient
+
+from typing import Callable, Dict, Optional
+from exasol.ds.sandbox.lib.logging import get_status_logger, LogType
+
+
+_logger = get_status_logger(LogType.DOCKER_IMAGE)
+
+
+def get_from_dict(d: Dict[str, any], *keys: str) -> str:
+    for key in keys:
+        if not key in d:
+            return None
+        d = d[key]
+    return d
+
+
+class ProgressReporter:
+    def __init__(self, verbose: bool):
+        self.last_status = None
+        self.verbose = verbose
+        self.need_linefeed = False
+
+    def _report(self, printer: Callable, msg: Optional[str], **kwargs):
+        if msg is not None:
+            printer(msg, **kwargs)
+
+    def _linefeed(self):
+        if self.need_linefeed:
+            self.need_linefeed = False
+            print()
+
+    def report(self, status: Optional[str], progress: Optional[str]):
+        if not self.verbose:
+            return
+        if status == self.last_status:
+            self._report(print, progress, end="\r")
+            self.need_linefeed = progress
+        else:
+            self.last_status = status
+            self._linefeed()
+            self._report(_logger.info, status)
```

Comment:
looks like this is re implementing features I would expect to be provided by the logging facility.

Reply:
As discussed